### PR TITLE
Add support for configuring tuned-main.conf file

### PR DESCRIPTION
**Pull request by jmencak**
This PR adds initial support for configuring values in global TuneD configuration file tuned-main.conf.  At this point, only `reapply_sysctl` configuration option is supported.

Users may now override host-provide sysctl values supplied in `/etc/sysctl{.conf,.d}`.

Also see rhbz#2039701.

### CODE REVIEW

**Comment by jmencak** [comment]
Note to reviewers.  To test the new functionality, I suggest the following CR:
```
apiVersion: tuned.openshift.io/v1
kind: Tuned
metadata:
  name: openshift-profile
  namespace: openshift-cluster-node-tuning-operator
spec:
  profile:
  - data: |
      [main]
      summary=Custom OpenShift profile
      include=openshift-node
      [sysctl]
      fs.inotify.max_user_instances=16384
    name: openshift-profile
  recommend:
  - match:
    - label: profile
    priority: 20
    profile: openshift-profile
    operand:
      debug: false
      tunedConfig:
        reapply_sysctl: false

```

Unless `reapply_sysctl: false`, you'll not be able to set `fs.inotify.max_user_instances`.

**Comment by jmencak** [comment]
@yanirq, I realize this will mean a new rebase for the controller-runtime refactoring.  I'm sorry about that, but we'll likely want to backport this to earlier branches as this resolves a BZ with an associated customer case.

**Comment by yanirq** [comment]
> @yanirq, I realize this will mean a new rebase for the controller-runtime refactoring. I'm sorry about that, but we'll likely want to backport this to earlier branches as this resolves a BZ with an associated customer case.

I will make sure to rebase and incorporate this change in the PR

**Comment by yanirq** [comment]
@jmencak Can an e2e test be added here ?  (under the basic test section could be enough)

**Comment by jmencak** [comment]
> @jmencak Can an e2e test be added here ? (under the basic test section could be enough)

It is on my list.  Need to think about a good scenario so that we don't get false positives and also do not break the cluster.

**Comment by dagrayvid** [comment]
@jmencak thanks for the PR. Overall it's looking good. One question, why is the `tunedConfig` section of the tuned CRD under the `recommend` section?  The recommend section is for "selection logic for all Tuned profiles." 

It seems to me that `tunedConfig` should be in its own section at the same level as `recommend` and `profile`.

**Comment by jmencak** [comment]
> @jmencak thanks for the PR. Overall it's looking good. One question, why is the `tunedConfig` section of the tuned CRD under the `recommend` section? The recommend section is for "selection logic for all Tuned profiles."
> 
> It seems to me that `tunedConfig` should be in its own section at the same level as `recommend` and `profile`.

Thank you for the review, and a good point, @dagrayvid. So, it is a past decision which may or may not have been a good one.  I see reasons for and against it.  The main reason is that it allows "per-profile/match" configuration for debugging and setting other operand configuration.  Think like: "I only want to enable debugging for profile 'A' that matches on a particular (set of) node(s) and leave the rest of the cluster alone".  That was the reason for having the `debug` option where it is.  Now, it is a question whether we want the same thing for other `TuneD` configuration.  As I said, I see reasons for and against, but this gives your more flexibility (at the expense of perhaps confusing users), so I went for it.  Unless you're dead set against it, I'd probably rather leave this as is.

File: pkg/operator/controller.go [hunk 1]
```diff
@@ -624,15 +625,22 @@ func (c *Controller) syncProfile(tuned *tunedv1.Tuned, nodeName string) error {
 		}
 	}
 
+	klog.Infof("syncProfile(): %v/%v: %v", profile.Spec.Config.TuneDConfig, operand.TuneDConfig, reflect.DeepEqual(profile.Spec.Config.TuneDConfig, operand.TuneDConfig))
+	klog.Infof("syncProfile(): profile.Spec.Config.TunedProfile=%v; tunedProfileName=%v", profile.Spec.Config.TunedProfile, tunedProfileName)
+	klog.Infof("syncProfile(): profile.Spec.Config.Debug=%v; operand.Debug=%v", profile.Spec.Config.Debug, operand.Debug)
+	klog.Infof("syncProfile(): profile.Spec.Config.ProviderName=%v; providerName=%v", profile.Spec.Config.ProviderName, providerName)
```
[the remark this answers is not in the document]
Comment (jmencak) [reply pkg/operator/controller.go]: Great catch!  Thank you David.  No, I meant to delete these.  Removed.

**Pull request merged**